src/main/resources/org/csanchez/jenkins/plugins/kubernetes/pipeline/samples/maven.groovy

```diff
@@ -55,7 +57,8 @@ public class SomeTest {
 }
         '''
             container('maven') {
-                sh 'mvn -B -ntp -Dmaven.test.failure.ignore verify'
+                // Maven needs write access to $HOME/.m2, which it doesn't have in the maven image because only root is a real user.
+                sh 'HOME=$WORKSPACE_TMP/maven mvn -B -ntp -Dmaven.test.failure.ignore verify'
```

Comment:
Arguably less intrusive to set `MAVEN_OPTS` as in my revised suggestion in https://github.com/carlossg/docker-maven/pull/460#issuecomment-2148418113 (or `HOME` may work too, did not try it, point being to handle that as part of the pod spec rather than overriding it in `sh` commands).